src/include/distributed/multi_copy.h

```diff
@@ -41,6 +41,13 @@ typedef struct CopyOutStateData
 
 typedef struct CopyOutStateData *CopyOutState;
 
+/* struct type to keep both hostname and port */
+typedef struct NodeAddress
```

Comment:
> We seem to have a couple of similar structures (WorkerNode, NodeConnectionKey). Is it worth reconciling with one of them, or are they too different in scope and purpose?

I considered to use them, but I think they are different in scope and purpose as you point. I'm slightly inclined to use this new type if you don't have any objections.